syntax = "proto3";

package zen.remote;

import "common.proto";

service SessionService
{
  rpc New(NewSessionRequest) returns (NewSessionResponse);

  rpc Ping(stream SessionPingRequest) returns (SessionTerminateResponse);
}

message NewSessionRequest {}

message NewSessionResponse
{
  uint64 id = 1;  // session id
}

message SessionPingRequest
{
  uint64 id = 1;  // session id
  bool done = 2;  // workaround for https://github.com/grpc/grpc/issues/10136
}

message SessionTerminateResponse {}
